crates/acir/src/optimiser/csat_optimiser.rs

```diff
@@ -144,8 +146,19 @@ impl Optimiser {
                     intermediate_gate.linear_combinations.push(left_wire_term);
                     intermediate_gate.linear_combinations.push(right_wire_term);
                     // Remove the left and right wires so we do not re-add them
-                    gate.linear_combinations.remove(x);
-                    gate.linear_combinations.remove(y);
+                    match x.cmp(&y) {
+                        Ordering::Greater => {
+                            gate.linear_combinations.remove(x);
+                            gate.linear_combinations.remove(y);
+                        }
+                        Ordering::Less => {
+                            gate.linear_combinations.remove(y);
+                            gate.linear_combinations.remove(x);
+                        }
+                        Ordering::Equal => {
+                            gate.linear_combinations.remove(x);
```

Comment:
because they are the same, you cannot remove twice an element from a vector, the second remove will be another element (this solves a bug that was in the previous code).